### PR TITLE
Manually update Microsoft.Private.Intellisense to 8.0.0-preview-20230918.1

### PR DESCRIPTION
**Pull request by carlossanlop**
Similar to https://github.com/dotnet/source-build-reference-packages/pull/774

Ran this command:

```
generate.cmd/sh -p Microsoft.Private.Intellisense,8.0.0-preview-20230918.1 -t text
```

I also made sure to delete the old files.

Workaround for https://github.com/dotnet/runtime/issues/77004

Theoretically, this should unblock https://github.com/dotnet/runtime/pull/92255 and https://github.com/dotnet/runtime/pull/92254

@mthalman @ericstj

### CODE REVIEW

**Comment by carlossanlop** [comment]
Do I/we need to do something special to make sure this works for `main`, `release/8.0` and `release/8.0-rc2`?

**Comment by ericstj** [comment]
Did you forget to add Microsoft.Private.Intellisense.8.0.0-preview-20230828.1.csproj?  Your previous PR included a CSProj.

**Comment by carlossanlop** [comment]
@ericstj I actually explicitly removed that folder since we now have a new package version, but I might have misunderstood the purpose of deletion, because I can see the CI failure is expecting the files from that package to exist.

Do you know why it is still expected?

**Comment by carlossanlop** [comment]
Maybe it's because I haven't merged the two PRs that bump the expected package version in Versions.props in main and release/8.0. Is that why?

**Comment by ericstj** [comment]
I just realized that's the old version, probably something still sees it.  Lets find out why.

**Comment by ericstj** [comment]
Here it is: https://github.com/dotnet/source-build-reference-packages/blob/2946b64e3384669e697ca7e71eb30ed5b25f6181/eng/Build.props#L27

**Comment by carlossanlop** [comment]
Thanks, @ericstj, good catch.

**Review by ViktorHofer — APPROVED** [review]

File: eng/Build.props [hunk 1]
```diff
@@ -24,7 +24,7 @@
       Format:
       <DependencyPackageProjects Include="$(RepoRoot)src\referencePackages\src\**\Microsoft.Extensions.Options.5.0.0.csproj" />
     -->
-    <DependencyPackageProjects Include="$(RepoRoot)src\textOnlyPackages\src\microsoft.private.intellisense\8.0.0-preview-20230828.1\Microsoft.Private.Intellisense.8.0.0-preview-20230828.1.csproj" />
+    <DependencyPackageProjects Include="$(RepoRoot)src\textOnlyPackages\src\microsoft.private.intellisense\8.0.0-preview-20230918.1\Microsoft.Private.Intellisense.8.0.0-preview-20230918.1.csproj" />
```
Comment (mthalman) [inline eng/Build.props:27]: @MichaelSimons - For a version upgrade like this, is this acceptable to change the version in Build.props like this?

**Comment by carlossanlop** [comment]
The backporting bot is not enabled in this repo so I'll do it manually.

**Pull request merged**